src/coord/rmr/rmr.c

```diff
@@ -475,6 +475,8 @@ int MRIteratorCallback_ResendCommand(MRIteratorCallbackCtx *ctx) {
 void MRIteratorCallback_ProcessDone(MRIteratorCallbackCtx *ctx) {
   short inProcess = __atomic_sub_fetch(&ctx->it->ctx.inProcess, 1, __ATOMIC_RELEASE);
   if (!inProcess) {
+    // Unblock the channel before calling `ProcessDone`, as it may trigger the freeing of the iterator
+    MRChannel_Unblock(ctx->it->ctx.chan);
     MRIterator_Release(ctx->it);
```

Comment:
1. Comment is incorrect
2. As discussed, this is a race condition and we should unblock the channel after attempting to release the iterator (setting the freeing flag). IMO the call to unblock should be part of the release flow, but we can also make the `MRIterator_Release` call return a value to indicate whether we freed the iterator or not, and call the unblock accordingly